File: robotframework/src/coffee_machine.py
class CoffeeMachine:
    def __init__(self):
        self.started = False

        # Yes it's a magic machine :)
        self.fill_tank()
        self.fill_beans()
        self.empty_grounds()

        self._last_coffee_success = False

    def start(self, lang = 'en'):
        self.started = True
        self.lang = lang

    def stop(self):
        self.started = False

    @property
    def messages(self):
        i18n = {
            'en': {
                'tank': 'Fill tank',
                'beans': 'Fill beans',
                'grounds': 'Empty grounds',
                'ready': 'Ready'
            },
            'fr': {
                'tank': 'Remplir reservoir',
                'beans': 'Ajouter grains',
                'grounds': 'Vider marc',
                'ready': 'Pret'
            }
        }

        return i18n[self.lang]

    @property
    def message(self):
        if not self.started:
              return ""

        if self.tank_content <= 10:
            return self.messages['tank']

        if self.beans_content < 3:
            return self.messages['beans']

        if self.grounds_content >= 30:
            return self.messages['grounds']

        return self.messages['ready']

    @property
    def coffee_served(self):
      return self._last_coffee_success

    def take_coffee(self):
        if (self.tank_content == 0 or self.beans_content == 0):
            self._last_coffee_success = False
        else:
            self._last_coffee_success = True

            self.tank_content -= 1
            self.beans_content -= 1
            self.grounds_content += 1


    def fill_tank(self):
        self.tank_content = 60

    def fill_beans(self):
        self.beans_content = 40

    def empty_grounds(self):
        self.grounds_content = 0

File: java-testng/src/main/java/com/testng/coffeemachine/CoffeeMachine.java
package com.testng.coffeemachine;

import java.util.HashMap;
import java.util.Map;

public class CoffeeMachine {

    private boolean started = false;
    public boolean coffeeServed = false;
    private int tankContent;
    private int beansContent;
    private int groundsContent;
    private String lang;

    public CoffeeMachine() {
        fillTank();
        fillBeans();
        emptyGrounds();
    }

    public void start(String lang) {
        started = true;
        this.lang = lang;
    }

    public void stop() {
        started = false;
    }

    public String message() {
        if(!started) return "";
        if (tankContent <= 10) return i18n("tank");
        if (beansContent < 3) return i18n("beans");
        if (groundsContent >= 30) return i18n("grounds");
        return i18n("ready");
    }

    public void takeCoffee() {
        if (tankContent == 0 || beansContent == 0) {
            coffeeServed = false;
        } else {
            coffeeServed = true;
            tankContent -= 1;
            beansContent -= 1;
            groundsContent += 1;
        }
    }

    public void fillTank() {
        tankContent = 60;
    }

    public void fillBeans() {
        beansContent = 40;
    }

    public void emptyGrounds() {
        groundsContent = 0;
    }

    private String i18n(String key) {
        Map<String,String> map = new HashMap<String, String>();
        if (lang.equals("fr")) {
            map.put("tank", "Remplir reservoir");
            map.put("beans", "Ajouter grains");
            map.put("grounds", "Vider marc");
            map.put("ready", "Pret");
        } else {
            map.put("tank", "Fill tank");
            map.put("beans", "Fill beans");
            map.put("grounds", "Empty grounds");
            map.put("ready", "Ready");
        }
        return map.get(key);
    }

}
